internal/cephfs/nodeserver.go

```diff
@@ -711,8 +744,38 @@ func (ns *NodeServer) NodeGetVolumeStats(
 	}
 
 	if stat.Mode().IsDir() {
-		return csicommon.FilesystemNodeGetVolumeStats(ctx, ns.Mounter, targetPath, false)
+		res, err := csicommon.FilesystemNodeGetVolumeStats(ctx, ns.Mounter, targetPath, false)
+		if err != nil {
+			return nil, err
+		}
+
+		return res, nil
 	}
 
 	return nil, status.Errorf(codes.InvalidArgument, "targetpath %q is not a directory or device", targetPath)
 }
+
+func (ns *NodeServer) startHealthChecker(ctx context.Context, stagingTargetPath string) {
+	hcDir, err := getHealthCheckPath(stagingTargetPath)
+	if err != nil {
+		log.WarningLog(ctx, "failed to identify health-check path %q: %v", stagingTargetPath, err)
+	}
+
+	err = ns.healthChecker.StartChecker(hcDir)
+	if err != nil {
+		log.WarningLog(ctx, "failed to start health-checker for path %q: %v", hcDir, err)
+	}
+}
+
+func getHealthCheckPath(stagingTargetPath string) (string, error) {
+	dir := path.Join(stagingTargetPath, ".csi")
+	err := os.Mkdir(dir, 0o700)
```

Comment:
`os.Mkdir()` or any other check that the directory on the CephFS mountpoint exists is problematic. The NodeGetVolumeStats call hangs when there is no available MDS (to cause intentional failures).

The directory for the timestamp should be created/checked once, when the health-checker is started (just after mounting, likely to succeed).